docs/notebooks/hydrotel.ipynb

```diff
@@ -0,0 +1,326 @@
+{
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "id": "d2be7c8a-3430-4bad-ae97-bff0fd631a89",
+      "metadata": {},
+      "source": [
+        "# Modelling module\n",
+        "## HYDROTEL\n",
+        "The `xhydro.modelling.Hydrotel` class is used to prepare and update configuration files for HYDROTEL, validate the project directory and the meteorological inputs, execute the model, and reformat the outputs to be more inline with CF conventions and other functionalities of `xhydro`.\n",
+        "\n",
+        "<div class=\"alert alert-info\"> <b>INFO</b>\n",
+        "`xhydro` does not distribute the HYDROTEL executable. If you want to use this module, you will have to get access to it first.\n",
+        "</div>\n",
+        "\n",
+        "### Initialising the model\n",
+        "`xhydro` does not prepare the project directory itself, which should be done beforehand. What the class does, when initiating a new instance of `xhydro.modelling.Hydrotel`, is allow modifying the entries located in the three main configuration files: `projet.csv`, `simulation.csv`, and `output.csv`. At minimum, when this class is initialised, the project folder will have to already exist and either 'default_options' will have to be True or SIMULATION COURANTE (current simulation) must be specified as a keyword argument in `project_options`.\n",
+        "\n",
+        "The arguments of the class are:\n",
+        "\n",
+        "- `project`: *str* or *os.Pathlike*. The full path to the project directory.\n",
+        "- `default_options`: *bool*. Whether to use default configuration options taken from `xhydro/modelling/data/hydrotel_defaults.yml`, or read them from the files on disk.\n",
+        "- `project_options`, `simulation_options`, `output_options`: *dict*. Configuration options to overwrite with new information.\n",
+        "\n",
+        "At any time after initialising the class, `Hydrotel.update_options()` can be called to update the three configuration files. When called, this function will overwrite the CSV files written on disk."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "25970397-024f-428f-92b6-1675a80e0161",
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "from pathlib import Path\n",
+        "from xhydro.modelling import Hydrotel\n",
+        "import xhydro as xh\n",
+        "\n",
+        "# Folder where to put the data\n",
+        "project_folder = Path().absolute() / \"_data\"\n",
```

Comment:
This line has a high likelihood of failing depending on how the notebooks are run: https://stackoverflow.com/questions/39125532/file-does-not-exist-in-jupyter-notebook

One approach I've taken to deal with this issue can be found here: https://github.com/Ouranosinc/xclim/commit/1dcf2adfef1056778c07938d6b91b17191395558